### PR TITLE
Allow same identifier in static and regular ops

### LINKED ISSUE #1097: Allow static operations to have the same name as regular operations
WebIDL currently specifies that:

> The identifier of a static operation also must not be the same as the identifier of a regular operation defined on the same interface.

In light of https://github.com/whatwg/fetch/issues/1389, I'd like to remove this limitation. The limitation was originally added in https://github.com/whatwg/webidl/commit/082298c0f7e4d231d121823778f820f7d852ca55 by @heycam. Does anyone remember why?

Is anyone opposed to this change?

### PR DESCRIPTION
**Pull request by lucacasonato**
Closes #1097

- [x] At least two implementers are interested (and none opposed):
   * Deno
   * Cloudflare Workers
   * Chromium
   * Mozilla
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   N/A
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1305399
   * Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1765992
   * Safari: https://bugs.webkit.org/show_bug.cgi?id=239672
   * Deno: already supported
   * Node.js: https://github.com/nodejs/node/issues/42831

***
<a href="https://whatpr.org/webidl/1098.html" title="Last updated on May 10, 2022, 10:13 AM UTC (2915f6b)">Preview</a> | <a href="https://whatpr.org/webidl/1098/ae0de45...2915f6b.html" title="Last updated on May 10, 2022, 10:13 AM UTC (2915f6b)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -2038,7 +2038,7 @@ as that of another operation on the interface, however.
 This is how operation overloading is specified.
 
 The [=identifier=] of a [=static operation=]
-also must not be the same as the identifier
+may be the same as the identifier
```
Comment (andreubotella) [inline index.bs:2041]: This paragraph used to be a normative restriction, but this change turns it into a clarification, so it should probably be a note.

Comment (domenic) [reply index.bs:2041]: Don't use "may" in a non-normative note; probably "can" is better here. https://infra.spec.whatwg.org/#conformance

**Review by Okiamgood — COMMENTED** [review]
Screenshot_20220305-200443
Screenshot_20220305-233720~2

**Comment by lucacasonato** [comment]
@TimothyGu / @EdgarChen would you mind taking a look? This is a prerequisite for https://github.com/whatwg/fetch/pull/1392.

**Review by EdgarChen — APPROVED** [review]
This looks good to me.
I checked how we handle the overloaded operations, [there are separated overload set for regular and static operation](https://webidl.spec.whatwg.org/#compute-the-effective-overload-set), so I think it is fine to have same identifier in regular and static operations.

**Comment by EdgarChen** [comment]
@TimothyGu, WDYT?

**Comment by EdgarChen** [comment]
BTW, I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1765992 for Gecko.

**Comment by lucacasonato** [comment]
Thanks @EdgarChen. Added the link to the PR description. I'll open a WebKit issue in a moment.

Can I count Gecko and Chromium as interested implementers for the working mode requirements?

**Comment by jasnell** [comment]
+1 to this. Definitely useful

**Comment by lucacasonato** [comment]
cc @tabatkins, as this may affect bikeshed IDL handling too.

**Comment by tabatkins** [comment]
Yeah, thanks for the ping. I will indeed need to make a change in how Bikeshed handles these. (The change makes sense tho, so Bikeshed shouldn't be a blocker here.)

**Comment by lucacasonato** [comment]
Support for this was merged in Chromium (https://chromium-review.googlesource.com/c/chromium/src/+/3623640), so we have two "interested implementers" now. @EdgarChen think we can land this?

**Comment by foolip** [comment]
I think this will impact MDN documentation, because static and instance members aren’t distinguished in the URL convention. Does anyone have a plan for that?

**Comment by domenic** [comment]
From what I understand MDN has had some consensus to move away from their current pretend-all-methods-are-static style for some time, but I haven't seen any actual updates: https://github.com/mdn/content/discussions/5121 and some other discussions.

**Comment by foolip** [comment]
It's been discussed, but there weren't any plans for anyone to work on it. I've commented on https://github.com/mdn/content/discussions/5121 and filed https://github.com/openwebdocs/project/issues/104 to put it on the radar, but I would expect `Response.json` to not be documented (or in BCD) until that's resolved.

**Pull request merged**